### ISSUE TITLE
Incorrect docstring, and Compressor

### ISSUE THREAD

**Issue by martindurant** [milesgranger/pyrus-cramjam#93]
`cramjam.lz4.Compressor`'s docstring is "Snappy Compressor object for streaming compression". 

I found this because I wanted to know whether the stream compressor is for the simple or block variant - both should be available, no? The output appears closer to the block version, but not identical.

Same comment for snappy, where I find Compressor is for the framed format, no raw variant.

**Comment by milesgranger** [comment]
Thanks for the note about incorrect docs. :+1:

As for Compressors, it depends what you want it for: (forgive me if I explain anything which you're already aware of, this also helps me)

- As implemented, they are only for framed/stream formats. This is because formats like snappy framed or gzip is composed of independently compressed pieces; 'raw/block' in snappy framed and deflate streams for gzip. Both of which will append or prepend additional information to headers and/or footers. [Same goes for Lz4](https://docs.fileformat.com/compression/lz4/#lz4-frame-format).  And also why none of the underlying crates implement [`Read`](https://doc.rust-lang.org/std/io/trait.Read.html)/[`Write`](https://doc.rust-lang.org/std/io/trait.Write.html) for block/raw formats, as we saw with the discussion in [snappy](https://github.com/BurntSushi/rust-snappy/issues/41#issuecomment-799826870)  and [lz4 missing Read/Write traits in block format](https://docs.rs/lz4/latest/lz4/block/index.html)

In summary, `Compressor` maintains the state until complete, then wrapping up any header/foot annotations for these formats.

- Now do you want a `Compressor` for these block/raw formats to [avoid repeated instantiations](https://github.com/milesgranger/pyrus-cramjam/blob/c6582ea20b5005d4dd6fb0965d0f2fcc97e919f2/src/snappy.rs#L78) of underlying encoders/decoders in some hot code path?  

This might be worth looking into if this is your aim. I can try to set aside some time to benchmark it a bit to see if exposing some `Encoder`/`Decoder` for block/raw formats would be beneficial in such a situation. But it serves a distinctly different purpose then existing `Compressor`s'.

**Comment by martindurant** [comment]
The hope was to avoid python bytes.join before passing data for compression, but it's a very minor thing. I am surprised, though. It should not require one frame per input to compress a stream (so long as the only flush() is the final one), using some internal buffer. But if the third party library doesn't provide the API, so be it!
Please don't devote any time to this.
